Make GCGLOwned a normal class

kkinnunen-apple opened the pull request:
#### a60ae9fc67214cc3a2bec47d8802cda6e3504b4a
<pre>
Make GCGLOwned a normal class
<a href="https://bugs.webkit.org/show_bug.cgi?id=268364">https://bugs.webkit.org/show_bug.cgi?id=268364</a>
<a href="https://rdar.apple.com/121913559">rdar://121913559</a>

Reviewed by Dan Glastonbury.

Make the GCGLOwned* classes easier to read and use by using a template
instead of preprocessor generation. This way the editors find the
definitions of GCGLOwned*.

* Source/WebCore/platform/graphics/GraphicsContextGL.h:
(WebCore::destroyFunc):
(WebCore::GCGLOwned::operator PlatformGLObject const): Deleted.
(WebCore::GCGLOwned::leakObject): Deleted.
(WebCore::GCGLOwned::~GCGLOwned): Deleted.

Canonical link: <a href="https://commits.webkit.org/273735@main">https://commits.webkit.org/273735@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c6fc21bf9a5051d19d1a4b0c6b2000981f457ede

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36543 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15480 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38764 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39205 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32774 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/37777 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17954 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12562 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/31391 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37103 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13049 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32324 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11418 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/11446 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32584 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40450 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33114 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32922 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37368 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11685 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9528 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35473 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13375 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8274 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12115 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12571 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg reviewed (approved):
:shipit: 

I think I used macros because I lack `templatefu`

kkinnunen-apple committed:
Make GCGLOwned a normal class
https://bugs.webkit.org/show_bug.cgi?id=268364
rdar://121913559

Reviewed by Dan Glastonbury.

Make the GCGLOwned* classes easier to read and use by using a template
instead of preprocessor generation. This way the editors find the
definitions of GCGLOwned*.

* Source/WebCore/platform/graphics/GraphicsContextGL.h:
(WebCore::destroyFunc):
(WebCore::GCGLOwned::operator PlatformGLObject const): Deleted.
(WebCore::GCGLOwned::leakObject): Deleted.
(WebCore::GCGLOwned::~GCGLOwned): Deleted.

Canonical link: https://commits.webkit.org/273735@main
